src/hotspot/share/gc/g1/g1ServiceThread.hpp

```diff
@@ -103,6 +105,9 @@ class G1ServiceThread: public ConcurrentGCThread {
   Monitor _monitor;
   G1ServiceTaskQueue _task_queue;
 
+  G1RemSetSamplingTask* _remset_task;
+  G1PeriodicGCTask* _periodic_gc_task;
```

Comment:
I would prefer them being embedded directly, `G1RemSetSamplingTask _remset_task;`, since they share the same lifespan with the enclosing object. No strong opinion; up to you.

Reply:
Me too, but then I would have to move the declaration of the tasks into the header. Now the above forward declaration is enough and I preferred that even more :)